analyzer.go

```diff
@@ -258,12 +262,32 @@ func (gosec *Analyzer) load(pkgPath string, conf *packages.Config) ([]*packages.
 		}
 	}
 
-	// step 3/3 remove build tags from conf to proceed build correctly.
+	return packageFiles, nil
+}
+
+func (gosec *Analyzer) load(pkgPath string) ([]*packages.Package, error) {
```

Comment:
Why do you need all these refactoring in the analyser? I would prefer that you only add the rule without changing the analyser since this is not in scope of this pull request. Please have a look at how is rule G109 implemented.

Reply:
The new rule needs to get the source code of the called function, namely the Rat.SetString source code.

Prior to this patch, it was not possible to load the source code of a called function from within a rule.

Now in the context is the `LoadPackages` function, which refers to the load function. It won't work without refactoring, since it's harder to get the context from inside the rule.

Reply:
Also, if GetPkgAbsPath returned an error, this does not mean that the package is not present, so you can continue executing the code

Reply:
`RawPkg *packages.Package` introduces a different level of abstraction in the analyser which is not required for this rule. Also this has performance issues in a large code base because you load all packages. 

You just need to check that the type of the variable where `SetString` is called is `big.Rat`. You can find an example on this rule https://github.com/securego/gosec/blob/34d144b3fa5d29af061157c1cd156a762f97f864/rules/slowloris.go#L51. 

I am not planning to merge the refactoring for this rule. Thanks for understanding!

Reply:
This was done in order not to check the runtime, but it's possible always output an issue